api spec for speech

PanosPeriorellis opened the pull request:
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

PanosPeriorellis committed:
api spec for speech

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-node

AutorestCI commented:
# Automation for azure-sdk-for-ruby

AutorestCI commented:
# Automation for azure-sdk-for-java

azuresdkci commented:
Can one of the admins verify this patch?

PanosPeriorellis commented:
This is not a patch. It is the first time upload of our swagger spec. I followed the example of the other cog services.

From: Azure Builder <notifications@github.com>
Sent: 21 March 2019 16:16
To: Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Panos Periorellis <Panos.Periorellis@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [Azure/azure-rest-api-specs] api spec for speech (#5445)


Can one of the admins verify this patch?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F5445%23issuecomment-475271286&data=02%7C01%7Cpanos.periorellis%40microsoft.com%7C521db1c7dd564cf752a108d6ae10206c%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636887781617549355&sdata=7clcBAihHRYXDRvmEGZLpwIaEaHBe4TXdyRtnaJK47Y%3D&reserved=0>, or mute the thread<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAYK9kh4hmsyOsVzTaVseYb6Guz3f9NV0ks5vY6IvgaJpZM4cBq_y&data=02%7C01%7Cpanos.periorellis%40microsoft.com%7C521db1c7dd564cf752a108d6ae10206c%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636887781617559364&sdata=fFs158CZOmP3N%2Bc7qRKV8nRPi7tgoXPZLGAvg2wJbA8%3D&reserved=0>.

dsgouda commented:
We have a separate process to onboard new RPs for the Net Sdk, please disregard `
Automation for azure-sdk-for-net ` related failures

PanosPeriorellis commented:
Ok can you sign off my PR then please.? Or tell me what I need to do

From: Deepak Shankargouda <notifications@github.com>
Sent: 21 March 2019 17:49
To: Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Panos Periorellis <Panos.Periorellis@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [Azure/azure-rest-api-specs] api spec for speech (#5445)


We have a separate process to onboard new RPs for the Net Sdk, please disregard Automation for azure-sdk-for-net related failures

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F5445%23issuecomment-475311631&data=02%7C01%7Cpanos.periorellis%40microsoft.com%7C8b8b4b40d6ec46ac67b308d6ae1d2956%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636887837606201795&sdata=2HDP7J8MYqJiLEl3DhODRbcVEcpx%2FPSltV3LFQRH870%3D&reserved=0>, or mute the thread<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAYK9klVgcixJ_tVRl4cD8q1Zrolkg4hzks5vY7gOgaJpZM4cBq_y&data=02%7C01%7Cpanos.periorellis%40microsoft.com%7C8b8b4b40d6ec46ac67b308d6ae1d2956%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636887837606211804&sdata=U%2FhAlk%2BYYxOMs8qVC7i49h%2BhBoTQn15JHSI3kGd1lbo%3D&reserved=0>.

nschonni commented on specification/cognitiveservices/data-plane/Speech/stable/v2.0/Speech.json:
```diff
@@ -0,0 +1,13590 @@
+                "properties": {
+                  "AcceptedLines": "11",
+                  "RejectedLines": "2",
+                  "ReportUri": "https://www.example.org/AcousticData/ed9b62f0-aa8e-419c-ab55-d7ca83c2f6b3?st=2018-02-09T18%3A07%3A00Z&se=2018-02-10T18%3A07%3A00Z&sp=r
```
occured -> occurred

nschonni replied:
levvel -> level

erhopf commented:
@shahabhijeet Do you know why the sdk-for-net is failing?

veronicagg reviewed (changes requested):
1. Given that this appears to be a completety new spec, has it been reviewed by the Azure API review board? 
2. Could you include someone else from your team to sing-off on the content of the PR?
3. In order to generate SDKs we need readme.md configuration files, please take a look at https://github.com/Azure/azure-rest-api-specs/tree/cc3861000c4cfe6b02c3dcd84ccb875d0968f4b3/specification/cognitiveservices/data-plane/CustomVision/Training and include these in the PR.
4. Are these APIs already deployed?

erhopf commented:
@veronicagg - Thanks for the response. 
1. As this is a production service and the docs are already published, the API review was waived. We've been working with Mike Kinsman to get this published. Here's the request: https://ceapex.visualstudio.com/Onboarding/_workitems/edit/72351
2. @PanosPeriorellis can review and sign-off on the content edits included in this PR. 
3. We'll review and update accordingly.
4. Yes. https://westus.cris.ai/swagger/ui/index#/

erhopf commented:
@veronicagg - Can you please link me to guidelines for creating the readme configuration files?

veronicagg commented:
@erhopf general guidance is at https://github.com/Azure/autorest/blob/87a5600e4af7ea185a887c63ad3153f0936fb0fa/docs/user/configuration.md
but I recommend that you copy form something similar to you, feel free to take a look at the latest PR I merged https://github.com/Azure/azure-rest-api-specs/pull/5164

PanosPeriorellis commented:
#sign-off

nschonni commented:
@PanosPeriorellis is there a reason for marking the typos as resolved without pushing anything? I know there is a bug with the GitHub preview on large files showing the wrong line on this view, but you can see the issues looking at the file view

PanosPeriorellis committed:
Update Speech.json

-corrected levvel-> level and occured -> occurred

AutorestCI commented:
# Automation for azure-sdk-for-js

AutorestCI commented:
# Automation for azure-sdk-for-go

PanosPeriorellis commented:
Sorry my mistake, I thought you had proposed changes which I was accepting. I have now performed the changes myself and pushed the updated version.

From: Nick Schonning <notifications@github.com>
Sent: 22 March 2019 08:24
To: Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Panos Periorellis <Panos.Periorellis@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Azure/azure-rest-api-specs] api spec for speech (#5445)


@PanosPeriorellis<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FPanosPeriorellis&data=02%7C01%7Cpanos.periorellis%40microsoft.com%7C494ffbaf13be4902f04708d6ae9752c9%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636888362319837006&sdata=DMYxdI8FWChUQFIDsksmzQ9gTGcikWU1rqLlb5pTPbs%3D&reserved=0> is there a reason for marking the typos as resolved without pushing anything? I know there is a bug with the GitHub preview on large files showing the wrong line on this view, but you can see the issues looking at the file view

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F5445%23issuecomment-475518917&data=02%7C01%7Cpanos.periorellis%40microsoft.com%7C494ffbaf13be4902f04708d6ae9752c9%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636888362319847003&sdata=2asOHwfISyWItBiYFnQgA5rrnhajG4jTX5AytO76JwQ%3D&reserved=0>, or mute the thread<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAYK9kiGFtn6a1EvtmFlJpgBYCjh1BBYhks5vZIUCgaJpZM4cBq_y&data=02%7C01%7Cpanos.periorellis%40microsoft.com%7C494ffbaf13be4902f04708d6ae9752c9%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636888362319847003&sdata=QRNDZkvb6KSVEJW8TORhmjo9CRViErQ%2Bgun0CkrNZ4U%3D&reserved=0>.

PanosPeriorellis committed:
added readme files

-added all read me files required following the example of FACE (seemed closer than custom vision)

PanosPeriorellis commented:
I added all the readme files that are required following the example from FACE

veronicagg reviewed (changes requested):
Please review CI errors for linter check https://github.com/Azure/azure-rest-api-specs/pull/5445/checks?check_run_id=82863674

veronicagg commented:
@shahabhijeet @dsgouda is the failure for .net generation due to being a new provider? should it be ignored?

dsgouda commented:
> 
> 
> @shahabhijeet @dsgouda is the failure for .net generation due to being a new provider? should it be ignored?

Yes, please ignore, we will on board them in the SDK repo separately

PanosPeriorellis commented:
+@Erik Hopf<mailto:Erik.Hopf@microsoft.com>

Guys, Is there anything else you need from us?  I have uploaded both the spec and the readme files.

From: Deepak Shankargouda <notifications@github.com>
Sent: 22 March 2019 22:04
To: Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Panos Periorellis <Panos.Periorellis@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Azure/azure-rest-api-specs] api spec for speech (#5445)


@shahabhijeet<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshahabhijeet&data=02%7C01%7Cpanos.periorellis%40microsoft.com%7C456c9c8e900440ed985c08d6af09f703%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636888854710327356&sdata=2eMtS0mxZBN2%2F09eFnNB08b3s95FhUquNRidpDSr9Lw%3D&reserved=0> @dsgouda<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdsgouda&data=02%7C01%7Cpanos.periorellis%40microsoft.com%7C456c9c8e900440ed985c08d6af09f703%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636888854710337354&sdata=2QiglYLE%2FWxj1UQq8ptiGN%2FYnxEsDiDbQr%2FCfFy3Cag%3D&reserved=0> is the failure for .net generation due to being a new provider? should it be ignored?

Yes, please ignore, we will on board them in the SDK repo separately

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F5445%23issuecomment-475782362&data=02%7C01%7Cpanos.periorellis%40microsoft.com%7C456c9c8e900440ed985c08d6af09f703%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636888854710337354&sdata=pr8YKS8Cjfx0cVtYGKOS5xQVJ8jGX2mZOWBo8nKMJG8%3D&reserved=0>, or mute the thread<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAYK9kjlp9vjwttphziUd0fxHOHPYEfWXks5vZUVYgaJpZM4cBq_y&data=02%7C01%7Cpanos.periorellis%40microsoft.com%7C456c9c8e900440ed985c08d6af09f703%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636888854710347352&sdata=QoBu5mvqYdoDyRCclcLsI1X%2BpfPRj0NvoY1w1BauxXE%3D&reserved=0>.

erhopf commented:
@veronicagg - What are the next steps? Do we owe you anything?

veronicagg commented:
@PanosPeriorellis @erhopf  yes, per my request of changes above, please review and correct CI errors for linter check https://github.com/Azure/azure-rest-api-specs/pull/5445/checks?check_run_id=82863674

PanosPeriorellis committed:
Update Speech.json

- fixed some errors related to "format": "uri"

PanosPeriorellis committed:
Update Speech.json

fixed anonymous type error

PanosPeriorellis committed:
updated speech.json with $ref

update 1 of 2 blokcs

PanosPeriorellis committed:
updated speech.json with $refs and camel naming notation

updated

PanosPeriorellis commented:
Hi Veronica,

I have resolved all issues but 1 and I need some help. The issue is the one below. The issue is the tool does not like our examples.



Rule

Location

Message

1

D5001 - XmsExamplesRequired<https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#D5001>

v2.0/​Speech.json:16<https://github.com/PanosPeriorellis/azure-rest-api-specs/blob/5e4303bd0652e43c7e4d1e3d80b31c62403aaf73/specification/cognitiveservices/data-plane/SpeechServices/stable/v2.0/Speech.json#L16>

Please provide x-ms-examples describing minimum/maximum property set for response/request payloads for operations.



Can someone from team help me with this. My question is should we re-write (re-format) our examples in the JSON? This would take some time, is there an alternative? Can we generate our docs without these examples?

Thanks
p

From: Veronica Giaudrone <notifications@github.com>
Sent: 25 March 2019 21:14
To: Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Panos Periorellis <Panos.Periorellis@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Azure/azure-rest-api-specs] api spec for speech (#5445)


@PanosPeriorellis<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FPanosPeriorellis&data=02%7C01%7Cpanos.periorellis%40microsoft.com%7C7903c71be7aa46d8355208d6b15e6fad%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636891416490632409&sdata=juTYKFW2LeiVjcPYgdYV0005HKHTg9dsectOlCwiJ0Q%3D&reserved=0> @erhopf<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Ferhopf&data=02%7C01%7Cpanos.periorellis%40microsoft.com%7C7903c71be7aa46d8355208d6b15e6fad%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636891416490632409&sdata=nR5dx1%2F5CrTq4AblTC%2B78vZuxh%2Bmv5XnJffEEFUmuv0%3D&reserved=0> yes, per my request of changes above, please review and correct CI errors for linter check https://github.com/Azure/azure-rest-api-specs/pull/5445/checks?check_run_id=82863674<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F5445%2Fchecks%3Fcheck_run_id%3D82863674&data=02%7C01%7Cpanos.periorellis%40microsoft.com%7C7903c71be7aa46d8355208d6b15e6fad%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636891416490642413&sdata=I%2BSVHH6nMu59se1vai%2B9OTq2h%2BJB7ikkA4oBGD%2Be9xI%3D&reserved=0>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F5445%23issuecomment-476359028&data=02%7C01%7Cpanos.periorellis%40microsoft.com%7C7903c71be7aa46d8355208d6b15e6fad%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636891416490652423&sdata=raq9KZbBpZpR29v5yA4IEJ76Wa6yKkr2aSIVzgxUkvs%3D&reserved=0>, or mute the thread<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAYK9krFyCkKmDKXybh1O9PNi2ArSB1mkks5vaS4NgaJpZM4cBq_y&data=02%7C01%7Cpanos.periorellis%40microsoft.com%7C7903c71be7aa46d8355208d6b15e6fad%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636891416490652423&sdata=eEFw%2FkrTRYKcx2m%2BsGN4H7t7yNj5mJ6houx%2FkPAKsuU%3D&reserved=0>.

veronicagg commented on specification/cognitiveservices/data-plane/SpeechServices/stable/v2.0/Speech.json:
```diff
@@ -1792,46 +1792,7 @@
-                  }
-                }
-              }
+              "$ref": "#/definitions/IReadOnlyDictionary`2"
```
is the "`" an intentional character in this name? When generating code, this name will turn into a Class name.

veronicagg commented on specification/cognitiveservices/data-plane/SpeechServices/stable/v2.0/Speech.json:
```diff
@@ -12296,6 +12184,48 @@
   },
   "definitions": {
+    "IReadOnlyDictionary`2": {
+    "title": "IReadOnlyDictionary`2",
```
minor: please indent the properties

veronicagg replied:
no need for Title here

veronicagg commented on specification/cognitiveservices/data-plane/SpeechServices/stable/v2.0/Speech.json:
```diff
@@ -12296,6 +12184,48 @@
+    "IReadOnlyDictionary`2": {
+    "title": "IReadOnlyDictionary`2",
+    "type": "object",
+    "properties": {
```
could you add descriptions for these properties?

veronicagg commented on specification/cognitiveservices/data-plane/SpeechServices/stable/v2.0/Speech.json:
```diff
@@ -12239,37 +12188,37 @@
     "type": "object",
     "properties": {
-      "None": {
+      "none": {
```
does the casing update reflect the service? Overall, we need to make sure the swagger reflects the service, so if updates made to the swagger due to issues reported by tools do not reflect the service, and the service can't be changed for this api version, you can ask for exceptions for the issues.

veronicagg reviewed (changes requested):
few comments inline

veronicagg commented on specification/cognitiveservices/data-plane/SpeechServices/stable/v2.0/Speech.json:
```diff
@@ -0,0 +1,13520 @@
+              ]
+            }
+          },
+          "401": {
```
please see https://github.com/Azure/azure-rest-api-specs/blob/da93a8f5a90e63c882d176021d6b935671faf0ba/documentation/creating-swagger.md#responses in particular https://github.com/Azure/azure-rest-api-specs/blob/da93a8f5a90e63c882d176021d6b935671faf0ba/documentation/creating-swagger.md#negative-responses
the recommendation is to have a "default" response for error responses which return an error model. If the spec describes each code, AutoRest for SDK code generation will have an "Object" return type for the operations, as "Error" and your actual return type have no other ancestor in common.

veronicagg commented on specification/cognitiveservices/data-plane/SpeechServices/stable/v2.0/Speech.json:
```diff
@@ -0,0 +1,13520 @@
+        ],
+        "parameters": [
+          {
+            "name": "id",
```
suggestion: you could abstract the parameter to the global parameter section, since you're referencing from multiple operations.

veronicagg commented on specification/cognitiveservices/data-plane/SpeechServices/stable/v2.0/Speech.json:
```diff
@@ -0,0 +1,13520 @@
+          "Custom Speech accuracy tests:"
+        ],
+        "summary": "Gets the list of accuracy tests for the authenticated subscription.",
+        "operationId": "GetAccuracyTests",
```
Operation ID should be of the form NOUN_VERB, the NOUN will become the "operation group" when generating SDK code, basically the name of the class in which all operations which shared it can be under. Ex. AccuracyTest_List

veronicagg commented on specification/cognitiveservices/data-plane/SpeechServices/stable/v2.0/Speech.json:
```diff
@@ -0,0 +1,13520 @@
+          "Custom Speech accuracy tests:"
+        ],
+        "summary": "Creates a new accuracy test.",
+        "operationId": "CreateAccuracyTest",
```
AccuracyTest_Create ?

veronicagg commented on specification/cognitiveservices/data-plane/SpeechServices/stable/v2.0/Speech.json:
```diff
@@ -0,0 +1,13520 @@
+          "Custom Speech accuracy tests:"
+        ],
+        "summary": "Gets the accuracy test identified by the given ID.",
+        "operationId": "GetAccuracyTest",
```
AccuracyTest_Get

veronicagg commented on specification/cognitiveservices/data-plane/SpeechServices/stable/v2.0/Speech.json:
```diff
@@ -0,0 +1,13520 @@
+                "properties": {
+                  "AcceptedLines": "11",
+                  "RejectedLines": "2",
+                  "ReportUri": "https://www.example.org/AcousticData/ed9b62f0-aa8e-419c-ab55-d7ca83c2f6b3?st=2018-02-09T18%3A07%3A00Z&se=2018-02-10T18%3A07%3A00Z&sp=r
```
should this be tagged as long-running? - see https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-long-running-operation

veronicagg replied:
long running? see my comment above referring to this

veronicagg replied:
please review https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-enum Using this extension and choosing modelAsString: true, can avoid breaking changes in the future as more values get added. Please evaluate this for all enums used in the spec.

veronicagg reviewed (changes requested):
More comments inline.

veronicagg commented:
Please review warnings from tools as well https://github.com/Azure/azure-rest-api-specs/pull/5445/checks?check_run_id=85238457#user-content-SDK-Warnings

veronicagg commented:
@PanosPeriorellis regarding your question on examples, examples inline should also work, and in that case you could request an exception for the warning https://github.com/Azure/adx-documentation-pr/wiki/Swagger-Validation-Errors-Suppression (make sure you're logged in to Github and have your account linked to Azure org to see this page). 
We recommend our custom x-ms-examples extension, as it makes the spec more readable without the inline examples. 
Regarding the examples showing up in documentation, @erhopf can you confirm this is the case?

veronicagg commented:
Additionally, you may want to take a look at the generated code from the SDKs (comments with generated PRs posted in the comments) to see if at least one the languages look like what you would expect. Thanks!

PanosPeriorellis committed:
Update Speech.json

updated based on veronica's suggestions.

PanosPeriorellis committed:
Update Speech.json

updated the response codes

veronicagg commented on specification/cognitiveservices/data-plane/SpeechServices/stable/v2.0/Speech.json:
```diff
@@ -0,0 +1,13585 @@
+          "Custom Speech accuracy tests:"
+        ],
+        "summary": "Gets the list of accuracy tests for the authenticated subscription.",
+        "operationId": "AccuracyTests_List",
```
Please update to AccuracyTest_List to make it inconsistent with the rest of the methods, like AccuracyTest_Create, since it's causing SDK generation issues in Java for example https://dev.azure.com/azure-sdk/public/_build/results?buildId=17447&view=logs&jobId=ef1a6f21-3eb6-5439-fa7a-dcd9cf589dd1&taskId=7d38bc47-3f36-578f-2b95-a19eb7179ec4&lineStart=155&lineEnd=156&colStart=1&colEnd=1

PanosPeriorellis replied:
Btw, I will not expose this to customers just yet, or at least until we fix the error. I just want to complete this so we get on a roll, get to know your process and start making progress.

veronicagg replied:
@PanosPeriorellis The fact of merging it to master means it's exposed to customers. Why do you want to merge this PR to master in the current state?
If you're intending to auto-generate SDKs we need to make sure that what we merge, meets the quality bar for SDKs. Java SDK generation is currently failing due to the Model name inconsistencies and the way return codes are specified for negative responses are no good for SDK auto generation. I understand the APIs are in GA, but the changes requested affect SDK generation.

veronicagg reviewed (changes requested):
Thanks for addressing the comments @PanosPeriorellis . 
Regarding the suppression, since it needs to be fixed at some point, we won’t be suppressing (sorry about my previous comment). I will approve and merge with the error, once all the other comments are resolved. 
Regarding my comment about the error codes for operations, it looks like you’d like to go ahead with what you have, is that correct? 
I posted a couple of suggestions incline and here are some questions:
Please make sure to review warning and decide whether you want to fix them or not:
https://dev.azure.com/azure-sdk/public/_build/results?buildId=17447&view=logs&jobId=688669d0-441c-57c3-cf6d-f89a22ccfa5d&taskId=923b5a90-e462-5994-c166-86509c6bf5f2&lineStart=1111&lineEnd=1112&colStart=1&colEnd=1
I see errors reported from CI for:
Model validation for the examples inline the spec: https://dev.azure.com/azure-sdk/public/_build/results?buildId=17447&view=logs&jobId=8d2f9c49-cd83-5a9d-f3fe-2fd651afa742&taskId=43324f1f-6724-5d61-785f-f0ac9fd256a1&lineStart=63&lineEnd=64&colStart=1&colEnd=1

Once you confirm you’re done with changes and happy with the current state, I can approve and merge, please know that without addressing the error code responses comment, your SDKs will not have a good customer experience on return type for operations.
To request SDK releases please use  https://portal.azure-devex-tools.com/app/tools/request-api-release. If you change how things look in the future, be aware that it can result in breaking changes.

veronicagg commented:
@PanosPeriorellis I haven't heard back in more than 10 days on this PR, so I'll be closing it on Monday, and can be reopened when you get back to it. Thanks!

PanosPeriorellis commented:
Hey Veronica,

I have been following the discussion and your comments. My plan is to carry out the comments and the warnings that are left. But it would be after build.

If you want to close it then that’s ok. We can reopen it later in May.

Thanks for all the help
p

From: Veronica Giaudrone <notifications@github.com>
Sent: 26 April 2019 20:17
To: Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Panos Periorellis <Panos.Periorellis@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Azure/azure-rest-api-specs] api spec for speech (#5445)


@PanosPeriorellis<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FPanosPeriorellis&data=01%7C01%7Cpanos.periorellis%40microsoft.com%7Cb9a942d0dbed441e9c7a08d6ca736b73%7C72f988bf86f141af91ab2d7cd011db47%7C1&sdata=f%2BhbX6tXhSaYE2LoEVz7VgkB%2BmgI%2BDDzsbzG5rStoM4%3D&reserved=0> I haven't heard back in more than 10 days on this PR, so I'll be closing it on Monday, and can be reopened when you get back to it. Thanks!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F5445%23issuecomment-487152601&data=01%7C01%7Cpanos.periorellis%40microsoft.com%7Cb9a942d0dbed441e9c7a08d6ca736b73%7C72f988bf86f141af91ab2d7cd011db47%7C1&sdata=7%2BqfN0yVopkoQTSux5zQFErK9ROEPYndG1J7MfHe2Yg%3D&reserved=0>, or mute the thread<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAGBL3EVQFBSBGH7AQQ3BQRTPSNBK5ANCNFSM4HAGV7ZA&data=01%7C01%7Cpanos.periorellis%40microsoft.com%7Cb9a942d0dbed441e9c7a08d6ca736b73%7C72f988bf86f141af91ab2d7cd011db47%7C1&sdata=PlMq0wnHMlA5L8osLvvNwkuk3LDKyVzZB0EMehClPD8%3D&reserved=0>.

veronicagg commented:
@PanosPeriorellis sounds good, thanks for following up. Based on the timeline, I'll close it and you can tag me back when you're back to this. Thanks!
